tests/problems/space/test_alignment_problem.py

```diff
@@ -235,3 +235,23 @@ def test_pass_arguments(self, adata_space_rotate: AnnData, args_to_check: Mappin
         for arg, val in pointcloud_args.items():
             assert hasattr(geom, val)
             assert getattr(geom, val) == args_to_check[arg]
+
+    def test_alignment_order_preservation(self, adata_space_rotate: AnnData):
+        sc.pp.subsample(adata_space_rotate, fraction=0.99)
+        ap: AlignmentProblem = AlignmentProblem(adata=adata_space_rotate)
+        ap = ap.prepare(batch_key="batch", joint_attr={"attr": "X"})
+        ap = ap.solve(alpha=0.5, epsilon=1, rank=10)
+        ap.align(key_added="spatial_warped", mode="warp", reference="0")
+
+        threshold = 0.7
+        batch = "0"
+        mask = ap.adata.obs["batch"] == batch
+        aligned_x = ap.adata.obsm["spatial_warped"][mask, 0]
+        rank_aligned_x = np.argsort(aligned_x)
+        original_x = ap.adata.obsm["spatial"][mask, 0]
+        rank_original_x = np.argsort(original_x)
+        corr = np.corrcoef(rank_original_x, rank_aligned_x)[0, 1]
```

Comment:
what is the actual value for this example? then we can make it potentially tighter, thus closer to a regression test.

Reply:
It is 0.9999999999999999

Reply:
oh wow, isn't that suspiciously high?

Reply:
But I only check on the reference node. In fact I realized it should in fact stay the same maybe I should change it to equality then. And I checked they are equal at the moment. This is how we want the end result to look right? https://github.com/theislab/moscot/issues/801#issuecomment-2743353442

Reply:
Ok I updated the tests now it checks if the order is preserved or not. I also set `max_iterations=2` to speed it up since I realized this bug wasn't related to the transport matrix

Reply:
and yeah, I do agree that the reference batch remains the same, let's test for equality of the reference slide, and let's test for a high correlation for aligned slices.

Reply:
thats correct let me do that

Reply:
can we fix this please?